scripts/build/builders/host.py

```diff
@@ -483,6 +432,9 @@ def PostBuildCommand(self):
             self._Execute(['genhtml', os.path.join(self.coverage_dir, 'lcov_final.info'), '--output-directory',
                            os.path.join(self.coverage_dir, 'html')], title="HTML coverage")
 
+        if self.app.ExamplePath() == "java-matter-controller" and 'JAVA_PATH' in os.environ:
```

Comment:
why the environment test?

for the test use `self.app == HostApp.JAVA_MATTER_CONTROLLER`